pallets/xyk/src/lib.rs

```diff
@@ -521,21 +577,31 @@ impl<T: Trait> Module<T> {
 
         result
             .saturated_into::<u128>()
-            .saturated_into::<T::Balance>()
+            .saturated_into::<BalanceOf<T>>()
     }
 
     pub fn get_liquidity_asset(
-        first_asset_id: T::AssetId,
-        second_asset_id: T::AssetId,
-    ) -> T::AssetId {
+        first_asset_id: CurrencyIdOf<T>,
+        second_asset_id: CurrencyIdOf<T>,
+    ) -> CurrencyIdOf<T> {
+        //TODO get rid of unwraps
         if <LiquidityAssets<T>>::contains_key((first_asset_id, second_asset_id)) {
-            <LiquidityAssets<T>>::get((first_asset_id, second_asset_id))
+            <LiquidityAssets<T>>::get((first_asset_id, second_asset_id)).into()
         } else {
-            <LiquidityAssets<T>>::get((second_asset_id, first_asset_id))
+            <LiquidityAssets<T>>::get((second_asset_id, first_asset_id)).into()
         }
     }
 
     fn account_id() -> T::AccountId {
         PALLET_ID.into_account()
     }
+
+    pub fn balance(id: CurrencyIdOf<T>, who: T::AccountId) -> BalanceOf<T> {
```

Comment:
why do we need a wrapper for `total_balance` and `total_issuance` ?

Reply:
that was mainly for test purposes - moved that impl to mock module